src/main/java/io/fabric8/maven/docker/util/DockerMachine.java

```diff
@@ -0,0 +1,109 @@
+package io.fabric8.maven.docker.util;
+
+import java.io.File;
+import java.util.Map;
+
+import org.apache.maven.plugin.MojoExecutionException;
+
+import io.fabric8.maven.docker.AbstractDockerMojo;
+import io.fabric8.maven.docker.config.MachineConfiguration;
+
+/**
+ * get environment from DockerMachine
+ */
+public class DockerMachine {
+
+    private final Logger log;
+    private final MachineConfiguration machine;
+
+    public DockerMachine(Logger log, MachineConfiguration machine) {
+        this.log = log;
+        this.machine = machine;
+    }
+
+    /**
+     * Get the docker host url.
+     * <ol>
+     * <li>From &lt;dockerHost&gt; configuration</li>
+     * <li>From &lt;machine&gt; configuration</li>
+     * <li>From DOCKER_HOST environment variable</li>
+     * <li>Default to /var/run/docker.sock</li>
+     * </ol>
+     * @param dockerHost The dockerHost configuration setting
+     * @return The docker host url
+     * @throws MojoExecutionException
+     */
+    public String extractUrl(String dockerHost) throws MojoExecutionException {
```

Comment:
I am fine with name change.

Chas

> On Jun 26, 2016, at 6:35 PM, Roland Huß <notifications@github.com> wrote:
> 
> In src/main/java/io/fabric8/maven/docker/util/DockerMachine.java:
> 
> > +        this.machine = machine;
> > +    }
> > +
> > +    /**
> > +     * Get the docker host url.
> > +     * <ol>
> > +     * <li>From &lt;dockerHost&gt; configuration</li>
> > +     * <li>From &lt;machine&gt; configuration</li>
> > +     * <li>From DOCKER_HOST environment variable</li>
> > +     * <li>Default to /var/run/docker.sock</li>
> > +     * </ol>
> > +     * @param dockerHost The dockerHost configuration setting
> > +     * @return The docker host url
> > +     * @throws MojoExecutionException
> > +     */
> > +    public String extractUrl(String dockerHost) throws MojoExecutionException {
> I'm not so happy to put the general extraction of the Docker access URL in a class called DockerMachine when docker machine is only one of many ways to detect the url. If you dont mind I'll rename it to 'DockerUrlDetector` or so while integrating.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>